### ISSUE TITLE
[FEAT] prediction drift display output improvements

### ISSUE THREAD

**Issue by Nadav-Barak** [deepchecks/deepchecks#1800]
We would like to add additional information and improve visuality of the check additional output. based on [link](https://docs.google.com/document/d/1i2qE5J0GYXLN5L9a1D3c4J86BmLijONXrPkp1CjnPs0/edit)

for categorical drift, add the following information as hover to the graph:
1.  for each class select all examples from train on which the model predicted this class. Next, given a scorer, calculate the average error on  them.
2. The contribution of this class to the overall drift calculation (need some more DS baking)

(Hard) For regression drift, segments the data into bins based on prediction value, then do similar to what is mention above - calculate average train error on segment, calculate contribution to overall drift and display via hover on graph

**Comment by nirhutnik** [comment]
Out of the above suggestions, implement:
* Hover-on: add the contribution of that class to the drift

Do not implement:
* Hover-on: add the average score on this prediction class
** Why? Because this is prediction drift and we should not expect labels at this point. Moreover, performance on the specific label don't necessarily have a lot of meaning for the user, as drift here may also imply just a change in label distribution.
* Regression - segment data into bins
** Why? because no reason to hurt the display of the check just for the hover-on. also, because of the original reason. Moreover, categoricals have different drift methods, and trying to explain drift values "strength" in EMD by using bins is methodologically problematic.


### Logic:
* For drift "contribution" - 2 possible solutions:
** Simple - take relative change in each category to itself
** More accurate - re-run the drift method (if Cramer's V) in a "1-vs-all" manner. 
* In any way, these results should be normalized - either by them being summed to 1 or to the drift score. Would recommend to 1 and presenting them as estimated percentages.

### Notes:
* For PSI, this is actually easy as the drift score is an aggregation of different scores per category. So can just take the original results from function and use those (they even sum up nicely to the drift score).
* Because we normalize the categories' counts to 1 (as the datasets may not be of same size), this method is problematic. meaning, if I have 3 countries - A, B and C - and I get more samples from C because of a change there, then not only that C's counts will rise (causing drift), but A and B's counts will lower accordingly (e.g. - (0.3, 0.3, 0.4) changed to (0.2, 0.2, 0.6)), although the change was **only** in C.
* If we decide to implement, no reason not to implement for all kinds of drift.


### Tasks:
* Implement residuals drift percentages for categoricals, as mentioned above - 1 SP
* Add as hover in display - 0.5 SP
* Add tests - 0.5 SP

**Comment by Nadav-Barak** [comment]
We have an overall agenda in which we want to use prediction drift as a proxy to detect model detonation see [link](https://docs.google.com/document/d/1i2qE5J0GYXLN5L9a1D3c4J86BmLijONXrPkp1CjnPs0/edit). 
As such I think it will give a lot of value for users to whether or not the drift is likely to hurt the performance.

Good example is the ability to disguise between the following cases:
1. you are now predicting more labels / values in areas in which for the train data you performed very well
2. you are now predicting more labels / values in areas in which for the train data you performed very bad - this is potentially much worse

Even though I can understand your logical reasoning I think providing info regarding effect on model performance, even if partial, is better than none and the user will be happy to see it, Especially since it is the best we got for scenarios without labels in test.

**Comment by nirhutnik** [comment]
I undertstand this comes from this agenda.
But - 
1. Performance on train is not representative of test and can be misleading
2. Even so, performance on each category individually is not interesting. Discussed with Bressler - I think this should be solved by a new check that estimates performance based on predictions.

**Comment by Nadav-Barak** [comment]
1. I disagree with the premise. Training data is a reasonable estimator for performance on different segment of data in test set
2. It is not a complete solution but IMO better than None.

I agree that thinking on an additional check that would try and give an estimate of model deterioration is a good idea (but not sure if should be **instead** of this)

**Comment by nirhutnik** [comment]
1. If model is overfitted, the estimation will be very wrong. If you had validation set I would have agreed, but training alone is prone to huge errors.
2. Better than none doesn't mean worth the time, which here would be ~2 SPs, not to talk about the additional calculation time. I think that a new check is far better than this, and will completely replace this as a solution so no reason to invest time in this.

**Comment by noamzbr** [comment]
## Bottom line
1. Add the contribution of each class to the drift in hover-on, as detailed in @nirhutnik s comment
2. Everything else will be covered by [#1908](https://github.com/deepchecks/deepchecks/issues/1908)
